### PR TITLE
[openshift-saas-deploy] always generate IMAGE_TAG

### PR DESCRIPTION
**Pull request by maorfr**
related to https://issues.redhat.com/browse/APPSRE-3265

we currently only generate IMAGE_TAG if it is a parameter in the openshift template and it is not explicitly specified in the saas file parameters. there are cases where we need to use the image tag to generate other parameters, such as REPO_DIGEST.

generally, since there are no real operations to generate the image tag, we can just always do it.

### CODE REVIEW

File: reconcile/utils/saasherder.py [hunk 1]
```diff
@@ -441,12 +441,11 @@ def _process_template(self, options):
                 try:
```
Comment (2uasimojo) [inline reconcile/utils/saasherder.py:441]: The comment on L437 could use some help. (Latent; I should have fixed it when I factored out _parameter_value_needed.)

**Review by 2uasimojo — COMMENTED** [review]
/lgtm

File: reconcile/utils/saasherder.py (hunk 1 above)
Comment (maorfr) [reply reconcile/utils/saasherder.py:441]: what would you change it to? (also, it's Sunday).

Comment (2uasimojo) [reply reconcile/utils/saasherder.py:441]: Probably just:

```python
            # This relies on IMAGE_TAG already being calculated.
```

Comment (maorfr) [reply reconcile/utils/saasherder.py:440]: done

**Review by apahim — APPROVED** [review]
lgtm

**Pull request merged**